clients/feeder/timeouts.go

```diff
@@ -154,3 +155,27 @@ func ParseTimeouts(value string) ([]time.Duration, bool, error) {
 	}
 	return timeouts, false, nil
 }
+
+func HTTPTimeoutsSettings(w http.ResponseWriter, r *http.Request, client *Client) {
+	switch r.Method {
+	case http.MethodGet:
+		fmt.Fprintf(w, "%s\n", client.timeouts.String())
+	case http.MethodPut:
+		timeoutsStr := r.URL.Query().Get("timeouts")
+		if timeoutsStr == "" {
+			http.Error(w, "missing timeouts query parameter", http.StatusBadRequest)
+			return
+		}
+
+		newTimeouts, fixed, err := ParseTimeouts(timeoutsStr)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		client.WithTimeouts(newTimeouts, fixed)
```

Comment:
Is there any case we have race condition here?

Reply:
I'm thinking about the case where this happens in the middle of a query loop. In that case there can be a scenario where old `curTimeout` is used while new `timeouts` slice is replaced, then out-of-range error can cause panic.

Reply:
Not entirely sure about it. `curTimeout` is always set to 0 when using `WithTimeouts`. You might be right. Added mutex just in case 👍

Reply:
used `atomic.Value` instead of mutex in the end